packages/advanced-logic/specs/payment-network-erc20-escrow-0.1.0.md

```diff
@@ -0,0 +1,55 @@
+# Additional interpretation for ERC20 Fee Proxy: smart escrow (WIP)
+
+**TODO: These specifications are work in progress and might change a lot before they are fully supported by the library.**
+
+Prerequisite: Payment Network "ERC20 With Fee" (see [here](./payment-network-erc20-fee-proxy-contract-0.1.0.md)).
+
+## Description
+
+This is an additional interpretation for the payment network "ERC20 With Fee", allowing the ERC20 payments to be locked on an escrow contract.
+
+The payment is made beforehand to the smart escrow, which locks the funds. The payer can then unlock the funds for a certain target date, after which the request issuer can withdraw the funds. In case of collaboration conflict, the payer can also decide to freeze the payment, which triggers a one-year period freeze. After a freeze, only the payer can withdraw the funds.
+
+The escrow contract pulls ERC20 tokens from the user, there should be no manual transfer.
+
+The escrow contract relies on a proxy when funds are withdrawn. The proxy is responsible for sending `TransferWithReferenceAndFee` events and process the payment to the issuer.
+
+The contract should only emit payment events when funds are unlocked or frozen by the payer.
+
+## Contract
```

Comment:
this is WIP right?
it misses details on the function to withdraw.

I also think the payee should be able to reject a payment, which would have the consequence of sending back the funds to the payer, with no delay

Reply:
That is a good point. should we priorotize this for the MVP1? or can we have it as a MVP2 feature?

Reply:
Good point for payment rejection !

For the function to withdraw, we used to only specify the functions that impact interpretation but here I agree that it's a critical function and it should be documented properly.